### ISSUE TITLE
Newb Question

### ISSUE THREAD

**Issue by ph3l1x** [AHAAAAAAA/PokemonGo-Map#2911]
Maybe I am being a total newb. First time trying out this script. But Here is my command line.

python runserver.py -a google -u myemail@address.com -p password1234 -st 10 -k AIzaSyD0Xql2meu5JfEYQRkp -k "someplace, WA" -H www.domain.com -P 80

runserver.py: error: arguments -u/--username, -l/--location, -st/--step-limit are required

am I not including those options?

**Comment by numb401** [comment]
you put a -k for your location instead of an -l

**Comment by DajeRoma** [comment]
should be 
python runserver.py -a google -u myemail@address.com -p password1234 -st 10 -k AIzaSyD0Xql2meu5JfEYQRkp **-l** "someplace, WA" -H www.domain.com -P 80

**Comment by SimpleCookie** [comment]
You require a step limit.
add -st 3 and it might work

**Comment by NoMotion** [comment]
Should also be an IP for `-H`. To host locally (only accessible to people on your network) `127.0.0.1` to host publicly `0.0.0.0`

**Comment by ph3l1x** [comment]
Wow, super simple mistake on my part. Thanks guys! I got it working.

**Comment by ph3l1x** [comment]
Does this have any logging? If you have it public is it possible to see any results like who is visiting it etc?

**Comment by ph3l1x** [comment]
Or any way that this could be setup for multiple users? Seems you have to have one interface per person or they will be changing everyones location around?

**Comment by Pymous** [comment]
You can have some logs using NGINX (and his access.log) if you're using it as a proxy.

And yes, you need one instance for each individual you wan't :)

**Comment by ph3l1x** [comment]
Would it be to complicated to make this work for multiple people?

**Comment by crckdns** [comment]
To make it view only just set up a server-only flag, let it fetch from MySQL database which your actual scanner is writing in to. Then just share the port to the server-only instance. That way no one can change it.

**Comment by NoMotion** [comment]
Just one process per user. Host each on different port. Supply your friend
with ip:port

On Jul 31, 2016 4:58 PM, "ph3l1x" <notifications@github.com> wrote:

Would it be to complicated to make this work for multiple people?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/AHAAAAAAA/PokemonGo-Map/issues/2911#issuecomment-236464968>,
or mute the thread

**Comment by ph3l1x** [comment]
Well I mean being able to make it more public like pokevision where multiple people can visit the website and use it. I have a private group of local people that we like to hunt with in my town and would be nice if it could be setup so I could share this with my team.

**Comment by NoMotion** [comment]
What's wrong with what I just stated?

**Comment by ph3l1x** [comment]
Then I would have to create 20 servers running on different ports... Would be much easier to maintain having 1 server running that allows multiple people at once to select their own location? Is that a really hard code change?

**Comment by bmhanson** [comment]
Granted I am a newb to this also, but the problem as I see it with that is each process will scan centered on a single location.  If you all want the same data then I would expect you could all use the same server/port and view it.  BUT if you want each person to have their own location center then you would need different processes.

My son and I were hunting together earlier and I gave him the url to connect to from his phone.  He saw the same thing I did with no problem.  But I made sure he was not moving the location around, etc.  

I am still working through all this as I just installed it last night.  I am looking into the benefits of a database, etc.  I am behind the curve as it were from most of the people here.

**Comment by ph3l1x** [comment]
Yes, but just curious if there is any way this could be setup to use a single process and allow multiple location points.

**Comment by NoMotion** [comment]
I have not dived into what web server the map is using, however the
functionally you describe we implementing sessions on the web server then
supplying a standalone database for all users connected.

Theoretically each time a user connects the web server would generate a
session and dedicate workers (pokemongo-map processes) to each session.

To draw the map the web server would also have to read from the same
database that all processes are writing to.

This is a large code change to make happen

**Comment by ph3l1x** [comment]
If I was willing to pay for it would you be willing to help me do this?

**Comment by NoMotion** [comment]
You don't need to pay lol. I am still new to the code and someone will
probably get around to it before I get the chance to. I say you make a
feature request for session capability to allow concurrent users on a
single database.

However it might be deemed outside the scope of the project. I think the
idea was to make small personal instances with the map, not another
pokevision
